### ISSUE TITLE
Typos and improvement in help

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#85]
I've decided to create this issue so in case there are typos or improvements to make in the help we can put everything here, so it doesn't interfere with the main testing process.

Right now I've found the following typo in `transaction list` help:
```
Only show received transactions (not in combination with -x, -n, -c, -b or -g). Only show received transactions (not in combination with -n, -c, -b or -g).```

Seems like almost the same text is included into this line twice, probably the second sentence is the old one.

**Comment by d5000** [comment]
First error (`token transfers`) was already changed in an earlier commit.

1. Please ignore this for now, it can't be changed.
1b. You're correct, this was an error, fixed it to `-p`.
2. `-o` works with all types of decks. The addition is only because dPoD decks have several P2TH addresses.
3. Don't agree because -o only lists P2TH addresses, the deck ID is only shown to identify the deck.
4. `-d` and `-q` are only used in some of the options, in the others they won't have effect, but as mentioned elsewhere `-d` always shows a traceback when there is an error.
5. (transaction list mention of speedup for block locators): Done.
6. (token balances) You're correct the current description fits the `-a` mode. I added the mode without `-a`.
7.  (change for -a flag) Makes sense, changed.

Fixes are in commit 709830a.

**Comment by buhtignew** [comment]
I've got all the above. Letting this issue open to add other small notes of the kind.

**Comment by buhtignew** [comment]
The help `token init` describes the `-a` flag as following:
` In combination with -s, store blockheights for all initialized tokens/decks.`
Probably the `-s` flag should be edited into `-c` flag in that description.

**Comment by d5000** [comment]
All done in commit 7166bef.

**Comment by buhtignew** [comment]
Regarding you commit https://github.com/slimcoin-project/pacli/commit/10dc4a0434bea95b2158a82f9e388aff246c7e09 mentioned [here](https://github.com/slimcoin-project/pacli/issues/102#issuecomment-2143284694) maybe we could add the description of `token list -s -p` and `token list -s -b` mode of use in the `token list` help.

**Comment by d5000** [comment]
Done, will be uploaded in the next update.

**Comment by buhtignew** [comment]
Regarding your reply [here](https://github.com/slimcoin-project/pacli/issues/104#issuecomment-2144231873), maybe we can include that info into the `token transfers` and `card list` help, specifically in the Usage mode section.
If I've got it right the above commands are not usable without the token_id, label or name, while from the help as we have it now it seems like they are.

**Comment by d5000** [comment]
You're correct, the TOKEN positional argument was missing as mandatory, I added it for the next update.

Second addition also makes sense, added it.

**Comment by buhtignew** [comment]
>Second addition also makes sense, added it.

I still don't understand the meaning of the word `cards` in the Name section of our `token transfers` help:
`pacli token transfers - List all existing transactions (cards, i.e. issues, transfers, burns) of a token.`

**Comment by d5000** [comment]
A card (short form for CardTransfer) is an unit used for any kind of token transaction (issuance, transfer or burn transaction). There can however be more than one card in one Slimcoin transaction. For example when you transfer tokens to two recipients, this will be counted as two cards/CardTransfers.

**Comment by buhtignew** [comment]
I've got the meaning now. Wouldn't it make sense to change `card` into `CardTransfer` in that sentence for those like me who wouldn't be enough conscious of the short form of the term?

**Comment by buhtignew** [comment]
The previous post has become too long. I'm opening a new one.
_
Maybe we can consider editing the line `Specify a multiplier for the reward..` into `Specify an integer multiplier for the reward.` in the help of the `pobtoken spawn` and `attoken spawn` command?
_
Since the name of the `deck_spawn` was renamed (as mentioned [here](https://github.com/slimcoin-project/pacli/issues/153#issuecomment-2262098672)) into `spawn` for the `pobtoken` and `attoken` groups, we can also edit `deck_spawn` into `spawn` in the `attoken span -h` output (there is still the old term in the Description section).

**Comment by buhtignew** [comment]
I'm beginning with another post, since the previous one was becoming too long to manage.
_
In agreement to [this](https://github.com/slimcoin-project/pacli/issues/175#issuecomment-2274715046) maybe we can add in the `proposal list -h` the same info already added for the `podtoken state -h` and `podtoken votes -h`: `If deck is not specified, the default dPoD deck is used.`

**Comment by d5000** [comment]
I'm currenlty working on the first long batch of suggestions (on a whole I agree with them and have modified the docstrings accordingly) and stumbled upon this, which is a misunderstanding:

> While testing pobtoken burn_coins command it was not enough clear for me where the burned amount goes if no deck is mentioned (see https://github.com/slimcoin-project/pacli/issues/138). Then I've realized the standard PoB deck is being used in that case.

Burning coins is "neutral", a burn transaction is valid for all PoB token decks. It even gives you proof-of-burn "score" for the slimcoin algorithm. It doesn't matter if you use the pacli command or `slimcoind burncoins`.

**Comment by d5000** [comment]
In the next update most suggestions of the "big" batches from August 1 and 8 were implemented, with the following exceptions:

- proposal period: was recently changed so it accepts also miniids and labels.
- proposal list issue: needs more checks.
- last proposal show issue: As it's already common in other commands I don't show exclusive flags in the command string of the Usage modes sections anymore. Doing this with every command consistently would be simply too complicated to explain and maintain.
- proposal list message about default dPoD deck: was already implemented before.
- dPoD instead of PoD: Help texts were modified. I'm considering changing the `PoDToken` class into `DPoDtoken`, but decision has not been taken still. According to class conventions in Python the class should start with an uppercase letter which looks here a bit strange.

**Comment by buhtignew** [comment]
This is my reply to your previous post. I'm going to begin working on the last one you've published soon after published this one.

 > Burning coins is "neutral", a burn transaction is valid for all PoB token decks. It even gives you proof-of-burn "score" for the slimcoin algorithm. It doesn't matter if you use the pacli command or slimcoind burncoins.

 Of course, I've realized it later on.

>  Another comment: In reality I like the token (deck) wording more, because token/deck would suggest that token is a complete synonym for deck. It is true that token commands can replace deck commands but the other way around is not true. I'll try to be consistent with that wording though.

Agreed

> at_address is used because it is a system parameter of this dictionary. It stands both for gateway and burn address.

I see now. Does it mean that it's not possible to change that parameter in the dictionary at all? If not so, i.e. if it's changeable, I think it would make sense to have it more meaningful both for the AT and PoB tokens. And also it's easy to pick up the wrong address from the `token show TOKEN -i` output because there is the `issuer` address  before in the list.

> When the label is first stored when deck is initialized, the code checks for local labels if they exist, and if not it puts out a message inviting the user to select the deck's label himself.

 I'm not sure it works like that for the moment or maybe I don't have clear the context it happens in. 
For instance I've just initialized the `test_issuerisgateway`, which was not initialized on my side, by running `deck init test_issuerisgateway` and my output was:
    ```
    Deck not named locally. Searching global deck name ...
    Using matching deck with global name test_issuerisgateway, with id: 1d4d470601e6359a1f3b7901d120620cd11741b316e610ad532c30f91053a49c
    Importing deck: 1d4d470601e6359a1f3b7901d120620cd11741b316e610ad532c30f91053a49c
    Importing P2TH address from deck.
    Trying to store global deck name test_issuerisgateway as a local label for this deck.
    Stored deck:
    Label: test_issuerisgateway
    Value: 1d4d470601e6359a1f3b7901d120620cd11741b316e610ad532c30f91053a49c
    Done.
    ```
So I was not asked to chose the label the deck's global label was chosen automatically as local label.
_
Seems like the replies to the following 3 proposal I've made are missing:

> In agreement to the issue #142 where the use of the global labels for the `pobtoken burn_coins` command with the `-i` flag (`pobtoken burn_coins 1 -i ATTokenNewSpec`, for instance) was implemented and since the local labels were already usable with that flag I'd suggest us to make some modification of the `-i` flag description and maybe of the extended form of the flag itself as well.
I.e. we can change `--idstr` into `--id_or_label` and its description can be changed from `ID of the Token or Deck you want to check compatibility with.` into `ID or label of the Token/Deck you want to check compatibility with`.

> [...] Maybe it would even make sense to add the `address balance` output as well.

> It seems like the local addresses' labels are not usable in case of the claim requests for the multiple receivers (for instance as mentioned in #165 it's not possible to run the `attoken claim johns_attoken2 c6a73f51fc3d336689e43c60a6715110486fca80b9940cafb38c6ddba0b90ddd -r "[usage_checked, old_default, new_after_refill9, fresh_after_refill9, flag-l]" -a "[0.7, 0.1, 0.01, 0.3, 0.02]"` command). I don't know whether it would make sense to additionally stress this fact in the help of the `attoken claim`, `pobtoken claim` and `podtoken claim` commands.

**Comment by d5000** [comment]
> Does it mean that it's not possible to change that parameter in the dictionary at all? 

It is not "impossible" but currently what's being output is the original structure of the object. For future versions there may be a "ELI5" interface.

> I'm not sure it works like described above for the moment

I see now, my answer there was misleading. Yes, the deck init stores the global name if possible and only if there is a conflicting local label then the message about storing a label manually is print out.

Regarding the three items you thought I'd not "answered":  I will not comment on all small decisions I do regarding these messages. Normally if I don't comment on them I've implemented them at least partly, not necessarily "exactly" like you proposed. These are not things we have to discuss every time!

But here are the answers:

1) Regarding --id_or_label: Positional arguments will normally not have "speaking" names. The other part was implemented.

2) balance output for address set was already implemented. 

3) Local labels for `attoken claim` were also already implemented.

Regarding the `podtoken electorate` message you proposed to output if the user uses the command incorrectly, I repeat my stance on this:

I'll _normally_ not provide an error message for users using the commands incorrectly. There are potentially thousands of possible messages that would have been implemented in this case.

In some cases however you already saw I did implement a message. This is the case when I have to catch a system error, where otherwise a long traceback is shown which will confuse the user. But then the messages will be very simple and generic. So in the case with `-e`and `-s` it's perfectly fine if I only provide one message for both cases. This message will **almost never been printed out**, because users will use `-e` or `-s` only if they want to see the state at a specific epoch or blockheight. Or why should they use those flags?

Yes I'm maybe a bit salty again but it's always the same problem with some of your proposals. I can't hold the hand of the user all the time if he's not willing to read about the correct usage, much less with the Fire/pacli package which is normally meant for advanced/technical users.

The same goes for the issue with the `-a` flag before an address.

> Another idea for the -s flag could be just ignoring it if used without the blockheight, thus providing the standard podtoken electorate output.

I've already commented on this in the podtoken electorate issue. Makes no sense because why should the user enter `-s` without any blockheight? The flag's "point" is to provide the interface to enter the blockheight.

**Comment by buhtignew** [comment]
>  I will not comment on all small decisions I do regarding these messages. Normally if I don't comment on them I've implemented them at least partly, not necessarily "exactly" like you proposed.

I've understood that you are using this approach, but in the three cases I've mentioned I haven't found the way to understand what have been changed, that's why I've asked.
For me it's important to understand it to test the new functionality or change in order to avoid us accidentally adding bugs where an improvement was intended.

> Regarding `--id_or_label`: Positional arguments will normally not have "speaking" names. The other part was implemented.

Would it make sense to document this change somewhere in the `pobtoken burn_coins` help by adding a hint about the possibility to use the label along with the token's id after the `-i` flag? (If not I'm afraid I'll pop up with the "improvement" proposal  again once I've forgot about this discussion :-) )

> Local labels for `attoken claim` were also already implemented.

Then I'm doing something wrong on my end, because when I try to run `attoken claim johns_attoken2 c6a73f51fc3d336689e43c60a6715110486fca80b9940cafb38c6ddba0b90ddd -r "[usage_checked, old_default, new_after_refill9, fresh_after_refill9, flag-l]" -a "[0.7, 0.1, 0.01, 0.3, 0.02]" ` I'm getting `Error: Receiver invalid: [
` error.

> For token transfers, the message was edited.

On my end if I run `token transfers -h`  I can still see `TOKEN can be a token (deck) ID or a label.` in Description. I think if we cut out the "a" before the "label" the message would become more clear.

> ...much less with the Fire/pacli package which is normally meant for advanced/technical users.

This was enlightening for me, TBH. I haven't thought about it that way so far. Thank you.

**Comment by buhtignew** [comment]
I'm opening this new post since the previous has become a bit long and not comfortable to manage anymore.
_
I'd like to suggest the following improvement of the `transaction list -h`:
We can edit this: 
```
pacli transaction list DECK [-o [ORIGIN_ADDRESS]] -b [-u]
pacli transaction list -b [-o [ORIGIN_ADDRESS]]
pacli transaction list DECK [-o [ORIGIN_ADDRESS]] -g [-u]
```
into this:
```
pacli transaction list -b [-o [ORIGIN_ADDRESS]]
pacli transaction list DECK [-o [ORIGIN_ADDRESS]] -b -u
pacli transaction list DECK [-o [ORIGIN_ADDRESS]] -g [-u]
```
That would improve the readibility of the help, IMO. Specifically the `-u` flag is not really optional in case of the `-b` flag usage with the DECK, because the `-b` flag alone (i.e. without the `-u` flag) should provide the same result for all the DECKs and also the same result of the `transaction list DECK -b` (that's why I'd suggest us to put the later usage mode before).

**Comment by buhtignew** [comment]
I'm creating this new post, since the previous one has become a bit long to manage.
_
When I run `token cache dt_111021 -v` on the deck that has been never cached I'm getting the following message: `This address was never cached.`, maybe we can edit it into `This deck was never cached.`?

**Comment by d5000** [comment]
> I was wondering whether the "TSLM" would change into "SLM" if I'd run on the main network.

Yes. The value is dynamic, it takes the Settings.network variable. Which also can be "PPC" and "TPPC" as Peercoin is also supported.

**Comment by buhtignew** [comment]
Great!
When you have time please have a look into the other typos: if I remember well some of them would be better to implement before the launch.

**Comment by buhtignew** [comment]
>>  IDK whether it would make sense to display the gateway address instead of issuer in the token list output, since when I need to find the token by the gateway address I don't know how to do and the issuer doesn't looks to me that relevant info in that context. But maybe we have a specific search tool already, haven't we?

> Changing that table specifically for AT tokens would be a major change and thus it would need a proper feature request. Not for this version.

Should I create another issue with the "next version" tag or put it somewhere in the wiki's feature requests?
_
If I run `attoken claim johns_attoken2 c6a73f51fc3d336689e43c60a6715110486fca80b9940cafb38c6ddba0b90ddd -r "[usage_checked, old_default, new_after_refill9, fresh_after_refill9, flag-l]" -a "[0.7, 0.1, 0.01, 0.3, 0.02]"` I'm still getting  `Error: Receiver invalid: [`.

**Comment by d5000** [comment]
Useful yes, but not necessary, because you have always the `deck show` variants. It would require that I rewrite the `deck list` command partially. And no, I'll first release the beta to see if there is any chance that the token platform really is used by someone, before wasting time on even more "luxury" features.

**Comment by d5000** [comment]
This is not a typo. The entries (dictionaries) with comma continue after the txid (they have a value_sent below), while in those without comma, the txid value is the last one.
Can i (or you) delete your last post? It's very long.

**Comment by buhtignew** [comment]
Got it and deleted as required.
